### PR TITLE
Support multiple request types

### PR DESCRIPTION
**Pull request by kevinchan159**
Add support for multiple request types for a path

### CODE REVIEW

File: src/AppDevRouter.js [hunk 1]
```diff
@@ -10,104 +12,106 @@ export type RequestType = 'GET' | 'POST' | 'DELETE';
 
 /**
  * AppDevResponse - the response from an HTTP request
- * 
+ *
  * Wraps a `success` field around the response data
  */
 class AppDevResponse<T> {
   success: boolean;
+
```
Comment (ilikesymmetry) [inline src/AppDevRouter.js:22]: why the empty line?

File: src/AppDevRouter.js [hunk 2]
```diff
@@ -10,104 +12,106 @@ export type RequestType = 'GET' | 'POST' | 'DELETE';
...
- * Subclasses can simply specify the HTTP method, the path, and a response 
- * hook to compute response data. This pattern is cleaner than raw Express 
+ * Subclasses can simply specify the HTTP method, the path, and a response
+ * hook to compute response data. This pattern is cleaner than raw Express
  * Router initialization with callbacks.
  */
 class AppDevRouter<T> {
   router: Router;
-  requestType: RequestType;
+
```
Comment (ilikesymmetry) [inline src/AppDevRouter.js:40]: same empty line thing

File: src/AppDevRouter.js (hunk 1 above)
Comment (kevinchan159) [reply src/AppDevRouter.js:22]: my linter added it saying `Expected new line between class members`

Comment (ilikesymmetry) [reply src/AppDevRouter.js:22]: fake news

Comment (kevinchan159) [reply src/AppDevRouter.js:22]: screen shot 2019-02-14 at 9 20 29 pm

Comment (young-k) [reply src/AppDevRouter.js:22]: i'm with @cswenberg i don't think this is standard, can you add to `.eslintrc` to ignore this rule?

File: src/AppDevRouter.js [hunk 3]
```diff
@@ -1,7 +1,9 @@
 // @flow
```
Comment (young-k) [inline src/AppDevRouter.js]: delete?

**Comment by young-k** [comment]
@meganle i think we should standardize indentation across all repos, is this something that we could do soon? either 2 or 4 is fine, but we should stick with one

File: src/AppDevRouter.js [hunk 4]
```diff
@@ -1,7 +1,8 @@
...
-
+import {
+    Router, Request, Response, NextFunction,
```
Comment (alanna-zhou) [inline src/AppDevRouter.js]: should this be alphabetized

File: src/AppDevRouter.js [hunk 5]
```diff
@@ -10,104 +11,104 @@ export type RequestType = 'GET' | 'POST' | 'DELETE';
...
+  requestTypes: Array<RequestType>;
 
   /**
    * Subclasses must call this constructor and pass in the HTTP method
    */
-  constructor(type: RequestType) {
-    this.router = new Router();
-    this.requestType = type;
+  constructor(types: RequestType) {
+      this.router = new Router();
```
Comment (alanna-zhou) [inline src/AppDevRouter.js]: I know that in Java it's bad to use "this"--is using "this" common style in Node? just curious

**Review by alanna-zhou — APPROVED** [review]
mostly just looks like formatting was changed, so it looks okay to me but dont quote me thx

File: src/AppDevRouter.js (hunk 4 above)
Comment (ilikesymmetry) [reply src/AppDevRouter.js]: oooh major key find alanna ! ! !

Comment (kevinchan159) [reply src/AppDevRouter.js]: you got me

File: src/AppDevRouter.js (hunk 5 above)
Comment (kevinchan159) [reply src/AppDevRouter.js]: is it bad in Java b/c its often not necessary? i'm not sure if its common in javscript but i don't think its something enforced

**Comment by meganle** [comment]
@young-k Let's do two spaces. I believe all the backend repos use this except the node transit backend which uses four @alanna-zhou @kevinchan159

File: src/AppDevRouter.js (hunk 5 above)
Comment (leewang-codes) [reply src/AppDevRouter.js]: it's pretty common in js

File: src/AppDevRouter.js [hunk 6]
```diff
@@ -1,7 +1,8 @@
 // @flow
-
+import {
+    NextFunction, Request, Response, Router,
```
Comment (young-k) [inline src/AppDevRouter.js]: this should be multiline, as in each import should be on its own line

**Review by young-k — COMMENTED** [review]
You also need to change this indentation to 2 spaces, and probably change the `.eslintrc` to reflect that as well

**Pull request merged**